inst/builtin/bs5/shiny/_rules.scss

```diff
@@ -70,11 +70,27 @@ $bslib-checkbox-radio-margin-right: 0.35em !default;
   }
 }
 
-// Apply box shadow effect to cards
+
 @mixin bslib-box-shadows {
-  .bslib-card {
+
+  // Add box-shadow & remove border from 'top-level' card()/layout_sidebar()
+  .bslib-card, .bslib-sidebar-layout {
     box-shadow: $box-shadow;
-    border: 0;
+    --#{$prefix}card-border-width: 0;
+    .bslib-card, .bslib-sidebar-layout {
+      box-shadow: none;
+      --#{$prefix}card-border-width: #{$card-border-width};
+    }
+    // If we did `@include color-mode(dark)` here,
+    // the selector wouldn't include the top-level @include
+    // selector scope, which is not what we want.
+    [data-#{$prefix}theme="dark"] & {
+      box-shadow: $box-shadow-dark;
+    }
+  }
```

Comment:
With your suggestion, we'd get the following selector:

```css
.bslib-page-sidebar > .bslib-sidebar-layout > .main [data-bs-theme="dark"] .bslib-card
```

instead of:

```css
[data-bs-theme="dark"] .bslib-page-sidebar > .bslib-sidebar-layout > .main .bslib-card
```


which is exactly the problem that the comment is trying to (yet failing) highlight (feel free to help improve it :))


Ultimately it doesn't feel like a huge deal if we don't respect `$color-mode-type` here since `input_dark_mode()` will (IIRC) mirror preferences on the data attribute, and that should cover 99%+ of use cases?

Reply:
Oh I see, we have to also take into account where you call `@include bslib-box-shadows`.

Yeah I think it's okay for us to the `data-bs-theme` attribute in this case. I think we should try to use the `color-mode()` mixin, unless it's extra cumbersome like in this case.